code-challenges.test.js

```diff
@@ -27,41 +27,110 @@ const numbersArray2 = [24, 27, 30, 33, 36]
-const object2 = { number: 0 }
-// Expected output: "0 is divisible by three"
-const object3 = { number: -7 }
-// Expected output: "-7 is not divisible by three"
+describe("div3" , () => {
+  it("takes a object as an argument and decides if the number inside it is evenly divisible by three", () => {
+  const object1 = { number: 15 }
+  // Expected output: "15 is divisible by three"
+  const object2 = { number: 0 }
+  // Expected output: "0 is divisible by three"
+  const object3 = { number: -7 }
+  // Expected output: "-7 is not divisible by three"
+  expect(div3(object1)).toEqual("15 is divisible by three")
+  expect(div3(object2)).toEqual("0 is divisible by three")
+  expect(div3(object3)).toEqual("-7 is not divisible by three")
+  })
+})
 
 // b) Create the function that makes the test pass.
-
-// Pseudo code:
+const div3 = (obj) => {
+  const number = obj.number
+  if (number % 3 === 0) {
+    return `${number} is divisible by three`
+  } else {
+    return `${number} is not divisible by three`
+  }
+}
+
+
+// Pseudo code: 
+// I need a function called "div3" that takes an object (obj) as input
+// I need to take a number from the object and assign it a variable called number
+// I need to make a function that checks if the number is divisible by three.
+// If the condition is true then return (Number) is divisible by three
+// otherwise it returns a string saying (Number) is not divisible by three
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+describe("capWords" ,() => {
+  it("an array of words and returns an array with all the words capitalized", () => {
+    const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+    // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
+    const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+    // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+  expect(capWords(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+  expect(capWords(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+  })
+})
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+const capWords = (words) => {
+  return words.map(word => word[0].toUpperCase() + word.slice(1))
+}
+
+
+// Pseudo code: 
+// I need a function called capWords that will take in words
+// have a return statement that maps the array
+// use .toUpperCase method to change each beginning letter using slice and the first index
```

Comment:
This pseudo code doesn't explain the concatenation. I would expand more on this one.